### PR TITLE
[REVIEW] codefresh: init at 0.74.8

### PR DESCRIPTION
**Pull request by jshcmpbll**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Is there a reason why the global nodePackages aren't used?

**Comment by jshcmpbll** [comment]
@SuperSandro2000  Just unfamiliarity to it. Ill look into it.

**Comment by jshcmpbll** [comment]
[build.log](https://github.com/NixOS/nixpkgs/files/5612839/build.log)

Errors I am getting when I try to build. :(

**Comment by jshcmpbll** [comment]
@SuperSandro2000 Following the steps to add a package to nodePackages there were a lot of changes to node-packages.nix. Is that expected and does that mean I have to then validate every nodepackage works after this change?

**Comment by SuperSandro2000** [comment]
> @SuperSandro2000 Following the steps to add a package to nodePackages there were a lot of changes to node-packages.nix. Is that expected and does that mean I have to then validate every nodepackage works after this change?
> 

Yeah, we need to update them all at once and no you don't need to test every package.

**Comment by SuperSandro2000** [comment]
@jshcmpbll please fix the merge conflict and eval error.

File: pkgs/development/node-packages/composition.nix [hunk 1]
```diff
@@ -6,12 +6,12 @@
 
 let
   nodeEnv = import ./node-env.nix {
-    inherit (pkgs) stdenv python2 util-linux runCommand writeTextFile;
+    inherit (pkgs) stdenv python2 utillinux runCommand writeTextFile;
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/composition.nix:9]: Please revert this.

File: pkgs/development/node-packages/node-env.nix [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # This file originates from node2nix
 
-{stdenv, nodejs, python2, util-linux, libtool, runCommand, writeTextFile}:
+{stdenv, nodejs, python2, utillinux, libtool, runCommand, writeTextFile}:
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/node-env.nix:3]: Please revert this.

File: pkgs/development/node-packages/node-env.nix [hunk 3]
```diff
@@ -396,7 +396,7 @@ let
     stdenv.mkDerivation ({
       name = "node_${name}-${version}";
       buildInputs = [ tarWrapper python nodejs ]
-        ++ stdenv.lib.optional (stdenv.isLinux) util-linux
+        ++ stdenv.lib.optional (stdenv.isLinux) utillinux
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/node-env.nix:399]: Please revert this.

File: pkgs/development/node-packages/node-env.nix [hunk 4]
```diff
@@ -470,7 +470,7 @@ let
         name = "node-dependencies-${name}-${version}";
 
         buildInputs = [ tarWrapper python nodejs ]
-          ++ stdenv.lib.optional (stdenv.isLinux) util-linux
+          ++ stdenv.lib.optional (stdenv.isLinux) utillinux
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/node-env.nix:473]: Please revert this.

File: pkgs/development/node-packages/node-env.nix [hunk 5]
```diff
@@ -516,7 +516,7 @@ let
     stdenv.mkDerivation {
       name = "node-shell-${name}-${version}";
 
-      buildInputs = [ python nodejs ] ++ stdenv.lib.optional (stdenv.isLinux) util-linux ++ buildInputs;
+      buildInputs = [ python nodejs ] ++ stdenv.lib.optional (stdenv.isLinux) utillinux ++ buildInputs;
```
Comment (SuperSandro2000) [inline pkgs/development/node-packages/node-env.nix:519]: Please revert this.

File: pkgs/tools/admin/codefresh/default.nix [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  yarnNix = ./yarn.nix;
+
+  src = fetchFromGitHub {
+    owner = "codefresh-io";
+    repo = "cli";
+    rev = "v${version}";
+    sha256 = "0rbrcplpc3ay951zj39738h6ny83h35a395k3li1mdji5kxvxmqz";
+  };
+
+  meta = with stdenv.lib; {
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/codefresh/default.nix:17]:
```suggestion
  meta = with lib; {
```

File: pkgs/tools/admin/codefresh/default.nix [hunk 7]
```diff
@@ -0,0 +1,24 @@
+{ yarn2nix-moretea, fetchFromGitHub, stdenv }:
```
Comment (SuperSandro2000) [inline pkgs/tools/admin/codefresh/default.nix:1]:
```suggestion
{ yarn2nix-moretea, fetchFromGitHub, lib }:
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
@jshcmpbll we have a merge conflict again. Can you give me a message on IRC when you resolved the merge conflict so that it does not happen again?

**Comment by jshcmpbll** [comment]
@SuperSandro2000 Merge conflict is fixed.

I dont know node well enough to trouble shoot this error I'm getting. I might see if a coworker of mine can help out but if you have any thoughts id appreciate it. My assumption was that it was just missing `faye-websocket` but even after adding that to `node-packages.json` I still get that error.

```npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but npm-shrinkwrap.json was generated for lockfileVersion@0. I'll try to do my best with it!
npm ERR! code ENOTCACHED
npm ERR! request to https://registry.npmjs.org/faye-websocket failed: cache mode is 'only-if-cached' but no cached response available.

npm ERR! A complete log of this run can be found in:
npm ERR!     /build/.npm/_logs/2021-01-20T09_39_31_341Z-debug.log

builder for '/nix/store/3v6g3l1kh86f2r9whk81dbhch4q9z552-node_codefresh-0.74.4.drv' failed with exit code 1
error: build of '/nix/store/3v6g3l1kh86f2r9whk81dbhch4q9z552-node_codefresh-0.74.4.drv' failed```

**Comment by SuperSandro2000** [comment]
Did you try regenerating the node packages nix file?

**Comment by jshcmpbll** [comment]
> Did you try regenerating the node packages nix file?

@SuperSandro2000 
I did, it just looks like this package has some weird and unique dependencies I wasn't aware of.  I have a workaround in place right now.. let me know if thats a valid solution - (the change to default.nix). If not im going to connect with some people over IRC and see if anyone can assist

**Comment by jshcmpbll** [comment]
@SuperSandro2000 Please review when you have time

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 105183` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package blacklisted:
</summary>
<ul>
<li>tests.nixos-functions.nixos-test</li>
</ul>
</details>
<details>

  <summary>1 package failed to build and are new build failure:</summary>
  <ul>
<li>netlify-cli: log <a href="https://termbin.com/izhh">https://termbin.com/izhh</a></li>
  </ul>
</details>
<details>
<summary>
49 packages built:
</summary>
<ul>
<li>antora</li>
<li>bitwarden-cli</li>
<li>castnow</li>
<li>epgstation</li>
<li>fast-cli</li>
<li>flood</li>
<li>joplin</li>
<li>lessc</li>
<li>mastodon-bot</li>
<li>mirakurun</li>
<li>pm2</li>
<li>postcss-cli</li>
<li>pulp</li>
<li>python37Packages.batchspawner</li>
<li>python37Packages.dockerspawner</li>
<li>python37Packages.jupyterhub</li>
<li>python37Packages.jupyterhub-ldapauthenticator</li>
<li>python37Packages.jupyterhub-systemdspawner</li>
<li>python37Packages.jupyterhub-tmpauthenticator</li>
<li>python37Packages.oauthenticator</li>
<li>python38Packages.batchspawner</li>
<li>python38Packages.dockerspawner</li>
<li>python38Packages.jupyterhub</li>
<li>python38Packages.jupyterhub-ldapauthenticator</li>
<li>python38Packages.jupyterhub-systemdspawner</li>
<li>python38Packages.jupyterhub-tmpauthenticator</li>
<li>python38Packages.oauthenticator</li>
<li>python39Packages.batchspawner</li>
<li>python39Packages.dockerspawner</li>
<li>python39Packages.jupyterhub</li>
<li>python39Packages.jupyterhub-ldapauthenticator</li>
<li>python39Packages.jupyterhub-systemdspawner</li>
<li>python39Packages.jupyterhub-tmpauthenticator</li>
<li>python39Packages.oauthenticator</li>
<li>redoc-cli</li>
<li>sloc</li>
<li>styx</li>
<li>thelounge</li>
<li>vimPlugins.coc-git</li>
<li>vimPlugins.coc-json</li>
<li>vimPlugins.coc-metals</li>
<li>vimPlugins.coc-prettier</li>
<li>vimPlugins.coc-rust-analyzer</li>
<li>vimPlugins.coc-stylelint</li>
<li>vimPlugins.coc-vetur</li>
<li>vscode-extensions.matklad.rust-analyzer</li>
<li>vscode-extensions.ms-python.vscode-pylance</li>
<li>vscode-extensions.vadimcn.vscode-lldb</li>
<li>wasm-text-gen</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

python37Packages.jupyterhub:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern ''npm', 'run', 'lessc', '--'' doesn't match anything in file 'setup.py'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>
python38Packages.jupyterhub:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern ''npm', 'run', 'lessc', '--'' doesn't match anything in file 'setup.py'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>
python39Packages.jupyterhub:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern ''npm', 'run', 'lessc', '--'' doesn't match anything in file 'setup.py'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by jshcmpbll** [comment]
....

**Comment by SuperSandro2000** [comment]
I am talking about the netlify build failure. The other things can be safely ignored.

> netlify-cli: log termbin.com/izhh

**Pull request closed without merging**